exercises/rational-numbers/test/test_rational_numbers.c

```diff
@@ -0,0 +1,439 @@
+   rational_t r2 = { 2, 3 };
+   rational_t expected = { 1, 3 };
+   rational_t actual = mul(r1, r2);
+   check_rational_numbers_equal(expected, actual);
+}
+
+static void test_multiply_positive_and_negative_rational_numbers(void)
+{
+   TEST_IGNORE();
+   rational_t r1 = { -1, 2 };
+   rational_t r2 = { 2, 3 };
+   rational_t expected = { -1, 3 };
+   rational_t actual = mul(r1, r2);
+   check_rational_numbers_equal(expected, actual);
+}
+
+static void test_multiply_two_negative_rational_numbers(void)
+{
+   TEST_IGNORE();
+   rational_t r1 = { -1, 2 };
+   rational_t r2 = { -2, 3 };
+   rational_t expected = { 1, 3 };
+   rational_t actual = mul(r1, r2);
+   check_rational_numbers_equal(expected, actual);
+}
+
+static void test_multiply_rational_number_by_reciprocal(void)
+{
+   TEST_IGNORE();
+   rational_t r1 = { 1, 2 };
+   rational_t r2 = { 2, 1 };
+   rational_t expected = { 1, 1 };
+   rational_t actual = mul(r1, r2);
+   check_rational_numbers_equal(expected, actual);
+}
+
+static void test_multiply_rational_number_by_one(void)
+{
+   TEST_IGNORE();
+   rational_t r1 = { 1, 2 };
+   rational_t r2 = { 1, 1 };
+   rational_t expected = { 1, 2 };
+   rational_t actual = mul(r1, r2);
+   check_rational_numbers_equal(expected, actual);
+}
+
+static void test_multiply_rational_number_by_zero(void)
+{
+   TEST_IGNORE();
+   rational_t r1 = { 1, 2 };
+   rational_t r2 = { 0, 1 };
+   rational_t expected = { 0, 1 };
+   rational_t actual = mul(r1, r2);
+   check_rational_numbers_equal(expected, actual);
+}
+
+static void test_divide_two_positive_rational_numbers(void)
+{
+   TEST_IGNORE();
+   rational_t r1 = { 1, 2 };
+   rational_t r2 = { 2, 3 };
+   rational_t expected = { 3, 4 };
+   rational_t actual = div(r1, r2);
+   check_rational_numbers_equal(expected, actual);
+}
+
+static void test_divide_positive_and_negative_rational_numbers(void)
+{
+   TEST_IGNORE();
+   rational_t r1 = { 1, 2 };
+   rational_t r2 = { -2, 3 };
+   rational_t expected = { -3, 4 };
+   rational_t actual = div(r1, r2);
+   check_rational_numbers_equal(expected, actual);
+}
+
+static void test_divide_two_negative_rational_numbers(void)
+{
+   TEST_IGNORE();
+   rational_t r1 = { -1, 2 };
+   rational_t r2 = { -2, 3 };
+   rational_t expected = { 3, 4 };
+   rational_t actual = div(r1, r2);
+   check_rational_numbers_equal(expected, actual);
+}
+
+static void test_divide_rational_number_by_one(void)
+{
+   TEST_IGNORE();
+   rational_t r1 = { 1, 2 };
+   rational_t r2 = { 1, 1 };
+   rational_t expected = { 1, 2 };
+   rational_t actual = div(r1, r2);
+   check_rational_numbers_equal(expected, actual);
+}
+
+static void test_absolute_value_of_positive_rational_number(void)
+{
+   TEST_IGNORE();
+   rational_t r = { 1, 2 };
+   rational_t expected = { 1, 2 };
+   rational_t actual = abs(r);
+   check_rational_numbers_equal(expected, actual);
+}
+
+static void
+    test_absolute_value_of_positive_rational_with_negative_numerator_and_denominator
+    (void) {
```

Comment:
Well this sucks

Reply:
The formatting in the example solution is weird too, when constructing the return values it doesn't look like it needs to be split over two lines.

Reply:
It shouldn't split the return according to line length.
Maybe it's doing that because it thinks `return (something) { blah }` looks like a function?